### ISSUE TITLE
[1.20.1] Cannot replace Terrestria or Regions Unexplored biomes

### ISSUE THREAD

**Issue by Cixon** [WerDei/Biome-Replacer#1]
### First off - great mod concept, I hope to see more mod creators utilize this!
I know this mod just came out but there's some incompatibilities with 2 mods I use:

Config:
cinderscapes:ashy_shoals > minecraft:basalt_deltas
regions_unexplored:magnolia_woodland > minecraft:cherry_grove
regions_unexplored:silver_birch_forest > terrestria:japanese_maple_forest

**Screenshots below show the biomes still appear (tested in new world)**

![Screenshot (119)](https://github.com/WerDei/Biome-Replacer/assets/69068153/585e6706-5d5b-4ee4-9ee6-a84122511217)

+-+-+-+-+-+-+-+-+-+-+-+-+-+-++-+-+-+-+-+-+-+-+-+

![Screenshot (116)](https://github.com/WerDei/Biome-Replacer/assets/69068153/f2c3f595-c5ed-4d55-8ec7-a2e228ebb79a)

**Comment by Cixon** [comment]
Mods in question

Cinderscapes:
https://legacy.curseforge.com/minecraft/mc-mods/cinderscapes

Regions Unexplored:
https://legacy.curseforge.com/minecraft/mc-mods/regions-unexplored

Terrestria:
https://legacy.curseforge.com/minecraft/mc-mods/terrestria

**Comment by Cixon** [comment]
### Tested on both fabric and forge
(forge mod [sinytra](https://legacy.curseforge.com/minecraft/mc-mods/sinytra-connector) which allows fml to use fabric mods)

Fabric - 0.14.22
FML - 47.1.3

Results varied on platforms. Sinytra needs more compat - this is unrelated issue.
As for **raw fabric testing**, cinderscapes appeared to be replaced with basalt deltas just fine.
Although, on both platforms we saw no differences made to Regions Unexplored biomes, this may be because of terrablender dependency getting in the way.

**Comment by WerDei** [comment]
Thank you for a report! Tested and confirmed the TerraBlender issue. Terrestria biomes are behaving strangely when replacing them too, even though Cinderscapes from same devs *seems* to work fine?
I'll look closer at both of these issues when I can.

**Comment by Cixon** [comment]
> Thank you for a report! Tested and confirmed the TerraBlender issue. Terrestria biomes are behaving strangely when replacing them too, even though Cinderscapes from same devs _seems_ to work fine?

iirc terrestria uses something called biolith which is a internal dependency similarly to terrablender although, cinderscapes doesn’t?

i could be wrong but both projects should be open-source.

> I'll look closer at both of these issues when I can.

awesome stuff! thank you for your speedy reply! also - sorry if i confused you with the forge stuff. i’m working on a fml modpack that uses fabric mods and figured i’d test if Sinytra is compatible; i reported it to be fixed so hopefully we can see 2 bugs ironed out for biome replacer soon! 🤞🤩

**Comment by WerDei** [comment]
Started rewriting the mod to support TerraBlender, but rand into some issues. I'm gonna shelve it for now, since I also realised that Biome Replacer isn't really needed for this - mods relying on TerraBlender or Biolith almost always let you configure and disable each biome individually.

Additionally, turns out that it's possible to replace datapack/vanilla biomes with these modded biomes, meaning it's only the other way around that does not work.

I'm going edit the mod description, clarifying that BiomeReplacer should only be used for datapack configuration, for now. I *might* fix this issue later, but no promises.

**Comment by Cixon** [comment]
> mods relying on TerraBlender or Biolith almost always let you configure and disable each biome individually.


Though this is true, biome replacer should mirror the removed biomes mass and rarity in the world. When you disable it, it closes the gap for other biomes to fill which isn’t the same. I find it better to replace a biome with your mod than to disable it and making a datapack for the biome I want more to then increase its spawn weight and size.

**Comment by WerDei** [comment]
Yeah, this is why I simply put the isue on hold while I got distracted. I just need to figure out how to cleanly catch the biome returned from any worldgen mod, hopefully without having to invade their code.

**Comment by LegendarySpy** [comment]
#21 This should hopefully be fixed _very very_ soon

**Comment by WerDei** [comment]
TerraBlender part of this has now been fixed, Biolith _might_ come later
